tests/TestBrowser.cpp

```diff
@@ -429,3 +430,57 @@ void TestBrowser::testValidURLs()
         QCOMPARE(Tools::checkUrlValid(i.key()), i.value());
     }
 }
+
+void TestBrowser::testBestMatchingCredentials()
+{
+    auto db = QSharedPointer<Database>::create();
+    auto* root = db->rootGroup();
+
+    // Test with simple URL entries
+    QStringList urls = {"https://github.com/loginpage", "https://github.com/justsomepage", "https://github.com/"};
+
+    auto entries = createEntries(urls, root);
+
+    browserSettings()->setBestMatchOnly(true);
+
+    auto result = m_browserService->searchEntries(db, "https://github.com/loginpage", "https://github.com/loginpage");
+    QCOMPARE(result.size(), 1);
+    QCOMPARE(result[0]->url(), QString("https://github.com/loginpage"));
+
+    result = m_browserService->searchEntries(db, "https://github.com/justsomepage", "https://github.com/justsomepage");
+    QCOMPARE(result.size(), 1);
+    QCOMPARE(result[0]->url(), QString("https://github.com/justsomepage"));
+
+    result = m_browserService->searchEntries(db, "https://github.com/", "https://github.com/");
+    m_browserService->sortEntries(entries, "github.com", "https://github.com/");
+    QCOMPARE(result.size(), 1);
+    QCOMPARE(result[0]->url(), QString("https://github.com/"));
+
+    browserSettings()->setBestMatchOnly(false);
+    result = m_browserService->searchEntries(db, "https://github.com/loginpage", "https://github.com/loginpage");
+    QCOMPARE(result.size(), 3);
```

Comment:
Because there is a perfect match in this query, it should be guaranteed to be returned first:

```suggestion
    QCOMPARE(result.size(), 3);
    QCOMPARE(result[0]->url(), QString("https://github.com/loginpage"));
```